module_info_import/models/module_information.py

```diff
@@ -49,59 +46,46 @@ def _get_module_information_vals_from_github_stat(self, module_info):
-        #        existing_module_names = test.keys()
-        #        for module_info in data:
-        #            if module_info['technical_name'] in existing_module_names:
-        #                module_id = test.get(module_info['technical_name'])
-        #                self.browse(module_id).write({'name': module_info['technical_name']})
-        #            else:
-        #                self.create({'name': {'name': module_info['technical_name']}})
-        odoo_version_dct = dict(
-            [(v.version, v.id) for v in self.env["odoo.version"].search([])]
-        )
+    def _update_modules(self, version, orga, repo, module):
+        odoo_version_dct = {
+            v.version: v.id for v in self.env["odoo.version"].search([])
+        }
 
-        for module_info in data:
-            module = self.search(
-                [("technical_name", "=", module_info["technical_name"])]
-            )
-            vals = self._get_module_information_vals_from_github_stat(module_info)
-            if module:
-                module.write(vals)
-                # create /update available versions of module.
-                if module_info["versions"]:
-                    version_modules = self.env["module.version"].search(
-                        [("module_id", "=", module.id)]
-                    )
-                    for version in module_info["versions"]:
-                        version_module = version_modules.filtered(
-                            lambda m: m.version_id.version == version
-                        )
-                        if version_module.state == "pending":
-                            version_module.state = "done"
-                        elif version_module:
-                            continue
-                        else:
-                            self.env["module.version"].create(
-                                {
-                                    "module_id": module.id,
-                                    "version_id": odoo_version_dct[version],
-                                    "state": "done",
-                                }
-                            )
+        # create Host orga & repo:
+        host_vals = {"organization": orga, "name": repo}
+        host_id = self.env["host.repository"].search([("name", "=", repo)], limit=1)
+
+        if host_id and host_id.organization != orga:
+            self.env["host.repository"].create(host_vals)
+        elif not host_id:
+            host_id = self.env["host.repository"].create(host_vals)
+
+        # check version
+        version_id = odoo_version_dct.get(version, "odoo_version_unknown")
+
+        vals = {
+            # "module_version_ids":
+            "available_version_ids": [(4, version_id, 0)],
+            "host_repository_id": host_id.id,
+            "technical_name": module,  # resultat du search
+        }
+
+        module = self.search([("technical_name", "=", module)])
+        if module:
```

Comment:
TODO we should simplify this code.

First we can introduce a method on module.module

```
def _should_update_module(self, version, orga):
    self.ensure_one()
    orga_priority = {"oca": 100, "akretion": 50}
    return (
        max(self.mapped("version_ids.name")) >= version
        and orga_priority.get(orga, 0) >= orga_priority.get(module.repo_id.orga, 0)
)


```